### PR TITLE
Flagged Posts and Recent Mentions

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Screens to show flagged posts and recent mentions.

Also added date separator for the search screen

Note: Need to update mattermost-redux before merging

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-10018

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux/pull/459)
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on:
* iOS 11.3 iPhone X Simulator
* Android 7.1.1 Emulator

#### Screenshots
iOS
simulator screen shot - iphone x - 2018-04-12 at 19 18 38

simulator screen shot - iphone x - 2018-04-12 at 19 19 15

simulator screen shot - iphone x - 2018-04-12 at 19 19 43

simulator screen shot - iphone x - 2018-04-12 at 19 20 11

simulator screen shot - iphone x - 2018-04-12 at 19 20 41

simulator screen shot - iphone x - 2018-04-12 at 19 21 17

simulator screen shot - iphone x - 2018-04-12 at 19 21 33

### CODE REVIEW

File: app/components/failed_network_action/index.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {TouchableOpacity, View} from 'react-native';
+
+import FormattedText from 'app/components/formatted_text';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import Cloud from './cloud';
+
+export default class NoResults extends PureComponent {
```
Comment (saturninoabril) [inline app/components/failed_network_action/index.js]: I think it's preferable to name this like `FailedNetworkAction`.

File: app/screens/flagged_posts/flagged_posts.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+                    onPermalinkPress={this.handlePermalinkPress}
+                    managedConfig={managedConfig}
+                    showFullDate={false}
+                />
+                {separator}
+            </View>
+        );
+    };
+
+    renderPostSeparator = () => {
```
Comment (saturninoabril) [inline app/screens/flagged_posts/flagged_posts.js]: Though it's convenient to have this per parent component, I feel like this should be pulled out into separate smaller component. I can see places where we can reuse this. Those that can be pulled out are:
- renderDateHeader
- renderEmpty
- renderPost
- renderPostSeparator

File: app/components/failed_network_action/index.js (hunk 1 above)
Comment (enahum) [reply app/components/failed_network_action/index.js]: oh yes my bad, I'll fix this

**Comment by enahum** [comment]
@saturninoabril by your first comment above it is now a prop instead that defaults to false

**Comment by saturninoabril** [comment]
Ah, you're right. I have overlooked it since I'm looking at your screenshot where search results still showing full date.
screen shot 2018-04-13 at 7 07 06 pm

**Comment by enahum** [comment]
@saturninoabril changes made

**Pull request merged**